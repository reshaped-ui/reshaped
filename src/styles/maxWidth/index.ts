import { responsiveClassNames, responsiveVariables } from "utilities/props";
import * as T from "styles/types";
import s from "./maxWidth.module.css";

const getMaxWidthStyles: T.DynamicStyleUtility<string | number> = (value) => {
	if (!value) return null;
	const variables = responsiveVariables("--rs-max-w", value);
	const classNames = responsiveClassNames(
		s,
		(value) => (typeof value === "number" ? "--type-unit" : "--type-literal"),
		value,
		{ excludeValueFromClassName: true }
	);

	return { classNames: [s.root, classNames], variables };
};

export default getMaxWidthStyles;
