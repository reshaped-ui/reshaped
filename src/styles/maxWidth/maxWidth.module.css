@responsive [style*="--rs-max-w-s:"] {
	@variable --rs-max-w none;
}

@responsive .--type {
	@value literal {
		max-width: var(--rs-max-w) !important;
	}

	@value unit {
		max-width: calc(var(--rs-max-w) * var(--rs-unit-x1)) !important;
	}
}
