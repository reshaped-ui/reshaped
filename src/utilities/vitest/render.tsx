import { render as vitestRender } from "vitest-browser-react";
import Reshaped from "components/Reshaped";
import "themes/reshaped/theme.css";

const Wrapper = (props: { children: React.ReactNode }) => {
	return <Reshaped>{props.children}</Reshaped>;
};

const render = (node: React.ReactNode) => {
	const screen = vitestRender(<div id="root">{node}</div>, {
		wrapper: Wrapper,
	});

	return { ...screen, container: document.getElementById("root")! };
};

export default render;
