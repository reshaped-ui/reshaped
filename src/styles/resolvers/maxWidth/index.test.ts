import { expect, test, describe } from "vitest";
import maxWidth from "./index";

describe("Styles/MaxWidth", () => {
	test("handles px value", () => {
		expect(maxWidth("50px")).toMatchSnapshot();
	});

	test("handles % value", () => {
		expect(maxWidth("50%")).toMatchSnapshot();
	});

	test("handles unit value", () => {
		expect(maxWidth(5)).toMatchSnapshot();
	});

	test("handles undefined value", () => {
		expect(maxWidth()).toMatchSnapshot();
	});

	test("handles responsive value", async () => {
		expect(maxWidth({ s: "50px", l: "50%" })).toMatchSnapshot();
	});
});
