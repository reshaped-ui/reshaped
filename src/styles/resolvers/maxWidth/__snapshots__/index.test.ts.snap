// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Styles/MaxWidth > handles % value 1`] = `
{
  "classNames": [
    "_root_255674",
    [
      "_--type-literal_255674",
    ],
  ],
  "variables": {
    "--rs-max-w-s": "50%",
  },
}
`;

exports[`Styles/MaxWidth > handles px value 1`] = `
{
  "classNames": [
    "_root_255674",
    [
      "_--type-literal_255674",
    ],
  ],
  "variables": {
    "--rs-max-w-s": "50px",
  },
}
`;

exports[`Styles/MaxWidth > handles responsive value 1`] = `
{
  "classNames": [
    "_root_255674",
    [
      "_--type-literal_255674",
      "_--type-literal--l_255674",
    ],
  ],
  "variables": {
    "--rs-max-w-l": "50%",
    "--rs-max-w-s": "50px",
  },
}
`;

exports[`Styles/MaxWidth > handles undefined value 1`] = `{}`;

exports[`Styles/MaxWidth > handles unit value 1`] = `
{
  "classNames": [
    "_root_255674",
    [
      "_--type-unit_255674",
    ],
  ],
  "variables": {
    "--rs-max-w-s": 5,
  },
}
`;
